federation/pkg/federation-controller/cluster/clustercontroller.go

```diff
@@ -148,9 +148,7 @@ func (cc *ClusterController) Run(stopChan <-chan struct{}) {
 }
 
 func (cc *ClusterController) GetClusterClient(cluster *federationv1beta1.Cluster) (*ClusterClient, error) {
-	cc.mu.RLock()
 	clusterClient, found := cc.clusterKubeClientMap[cluster.Name]
-	cc.mu.RUnlock()
```

Comment:
indeed, but some race can still happen if people uses this public function though. We probably want to make this private if it doesnt need to be exposed ?